docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+    m1oih6aevfcftysukvofwuth2bsuj5aahkdnpabscry7p7ljkgbxma.edgeql)
+
+
+.. note::
+
+    Squashing is still limited to schema changes, so queries inside 
+    data migrations will be discarded during a squash.
+
+EdgeDB Watch
+============
+
+Another option when quickly iterating over schema changes is ``edgedb watch``.
+This will create a long-running process that keeps track of every time you
+save as ``.esdl`` file inside your ``/migrations`` folder, letting you know
+if your changes have successfully compiled or not. The command itself will
+lead to the following input ``edgedb watch`` starts up:
+
+.. code-block::
+
+    Connecting to EdgeDB instance 'anything' at localhost:10700...
+    EdgeDB Watch initialized.
+    Hint: Use `edgedb migration create` and `edgedb migrate --dev-mode`
+    to apply changes once done.
+    Monitoring "/home/instancename".
+
+Unseen to the user, EdgeDB Watch will begin creating individual migration 
+scripts for every time you save a change to one of your files. These 
+are stored as separate "dev mode" migrations, which are sort of like 
+preliminary migrations that haven't been turned into a standalone 
+migration script yet.
+
+If you start with this schema:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        name: str;
+      }
+    }
+
+And then add a single property:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        name: str;
+        number: int32;
+      }
+    }
+
+You will see EdgeDB Watch quickly display "calculating diff" before 
+disappearing once the difference has been calculated and deemed to 
+be good.
+
+However, if you add incorrect syntax to the schema:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        name: str;
+        number: int32;
+        wrong_property: i32; # Should say int32, not i32
+      }
+    }
+
+Then EdgeDB Watch will suddenly pipe up and inform you that the schema 
+can't be resolved:
+
+.. code-block::
+
+    error: type 'default::i32' does not exist
+    ┌─ myproject/dbschema/default.esdl:5:25
+    │
+    5 │         wrong_property: i32;
+    │                         ^^^ error
+
+    Schema migration error:
+    cannot proceed until .esdl files are fixed
+
+Once you correct the ``i32`` property to ``int32``, EdgeDB Watch will 
+let you know that things are okay now, and will become quiet again:
+
+.. code-block::
+
+    Resolved. Schema is up to date now.
+
+EdgeDB Watch is best run in a separate tab on your command line so 
+that you can take care of other tasks—including officially migrating 
+when you are satisfied with your current schema—without having to 
+stop the process.
+
+If you are curious what is happening as EdgeDB Watch does its thing, 
+try the following query after you have made some changes:
+
+.. code-block::
+
+    group schema::Migration {
+        name,
+        script
+    } by .generated_by;
+
+Some migrations will contain nothing in their ``generated_by`` property, 
```

Comment:
Add a brief sentence after the query telling what it does. Maybe something like this: "This will return a list of applied migrations."

Reply:
https://github.com/edgedb/edgedb/pull/6656/commits/7c4494f93ec40c69bda1d94f6ec3eb072e1293a2

(A lowercased Edgedb watch slipped in there, reverted until we decide what to do with it)